src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts

```diff
@@ -292,43 +292,64 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
   }
 
   private getCaseFieldsFromCurrentAndPreviousPages(): CaseField[] {
-    const result: CaseField[] = []
-    this.wizard.pages.forEach ( page => {
+    const result: CaseField[] = [];
+    this.wizard.pages.forEach(page => {
       if (page.order <= this.currentPage.order) {
-        page.case_fields.forEach( field => result.push(field));
+        page.case_fields.forEach(field => result.push(field));
       }
-    })
+    });
     return result;
   }
 
   private buildCaseEventData(): CaseEventData {
-    // Get hold of the fields specific to the current page that we're going to submit.
-    const pageFormFields = this.formValueService.filterCurrentPageFields(this.currentPage.case_fields, this.editForm.value);
+    const formValue: object = this.editForm.value;
 
-    // Sort out the dynamic lists.
-    this.formValueService.sanitiseDynamicLists(this.currentPage.case_fields, pageFormFields);
+    // Get the CaseEventData for the current page.
+    const pageFields: CaseField[] = this.currentPage.case_fields;
+    const pageEventData: CaseEventData = this.getFilteredCaseEventData(pageFields, formValue, true);
 
-    // Get hold of the CaseEventData and immediately tidy it up.
-    const caseEventData: CaseEventData = this.formValueService.sanitise(pageFormFields) as CaseEventData;
-    this.formValueService.removeUnnecessaryFields(caseEventData.data, this.currentPage.case_fields, true);
+    // Get the CaseEventData for the entire form (all pages).
+    const allCaseFields = this.getCaseFieldsFromCurrentAndPreviousPages();
+    const formEventData: CaseEventData = this.getFilteredCaseEventData(allCaseFields, formValue, false, true);
 
-    // Now add the remaining bits and pieces to the CaseEventData,
-    // The event_data should be the full context of the event, including values from previous pages, but not labels
+    // Now here's the key thing - the pageEventData has a property called `event_data` and
+    // we need THAT to be the value of the entire form: `formEventData.data`.
+    pageEventData.event_data = formEventData.data;
```

Comment:
I assume this also resolves the issue mentioned in the previous ticket about DynamicList values?

Reply:
Yes, that's precisely what this is for. The `formEventData.data` property is all sanitised and has the dynamic list values sorted out. I've tested locally to make sure it's behaving itself and it all looks good to me. Will still need one of the QAs to confirm in their test harnesses, though.